webapp/graphite/finders/standard.py

```diff
@@ -26,6 +27,12 @@ def __init__(self, directories=None):
 
     def find_nodes(self, query):
         clean_pattern = query.pattern.replace('\\', '')
+
+        # translate query pattern if it is tagged
+        if ';' in query.pattern and not query.pattern.startswith('_tagged.'):
```

Comment:
Then the result could be kept for line 61, so it doesn't have to scan through the string on each iteration of the following `for`